wso2-synapse-unit-test-plugin/src/main/java/org/wso2/synapse/unittest/SynapseTestCaseFileReader.java

```diff
@@ -70,8 +72,32 @@ static String processArtifactData(String synapseTestCaseFilePath) {
 
             QName qualifiedTestCases = new QName("", Constants.TEST_CASES_TAG, "");
             OMElement testCasesNode = importedXMLFile.getFirstChildWithName(qualifiedTestCases);
-            Iterator<?> testCaseIterator = testCasesNode.getChildElements();
-            if (!testCaseIterator.hasNext()) {
+            if (testCasesNode != null) {
```

Comment:
this makes testCaseName mandatory ? if so, doesn't this change make it harder for users who use maven commands directly. Now they have to provide all the test case names. I think we need to support running all the test cases by default and use this new array to limit if needed.

Reply:
No this change does not make the test case name a mandatory field. The functionality will be the same unless a user passes a test case name as a parameter to the mvn test command. A condition is used in [1] to check whether a test case is given and if no test case has been given, then no test cases will be removed from the OMElement and therefore all the test cases in the unit test will be executed as usual.

[1] https://github.com/wso2/maven-tools/pull/145/files#diff-f17b6e674260e7bc276e52935557d4ca5c5e206a6f2fdc18b61e4d24b8b640edR81